app/pages/Converter/page.tsx

```diff
@@ -1,3 +1,294 @@
+      display: false,
+      ticks: {
+        display: false,
+      },
+    },
+    "y-axis-2": {
+      display: false,
+      ticks: {
+        display: false,
+      },
+    },
+  },
+};
+function Converter() {
+  const [selectedCoinsData, setSelectedCoinsData] = useState<TableDataProps[]>(
+    []
+  );
+  const [chartData, setChartData] = useState<{
+    prices: number[][];
+    dates: string[][];
+  }>({
+    prices: [[], []],
+    dates: [[], []],
+  });
+  const [coinAmount, setCountAmount] = useState<string>("");
+  const [swap, setIsSwap] = useState<boolean>(false);
+  const [days, setDays] = useState(["1", "7", "14", "30", "max"]);
+  const [selectedDays, setSelectedDays] = useState("7");
+  const dispatch = useDispatch<AppDispatch>();
+  const currentCurrency = useAppSelector(state => state.currency.currencies);
+  const selectedCoins = useAppSelector(state => state.selectCoin);
+  const coinsData = useAppSelector(state => state.coins.coins);
+  const currentDate = new Date();
+  const currentTime = currentDate.toLocaleTimeString("en-US", {
+    hour12: false,
+  });
+
+  const fetchChartData = async (coin: any, index: number) => {
+    try {
+      const chartData = await dispatch(
+        fetchGraphData({
+          currency: currentCurrency,
+          name: coin,
+          days: selectedDays,
+          daily: selectedDays === "1" ? "" : "daily",
+        })
+      );
+      const prices = chartData.payload.prices.map(
+        (price: [number, number]) => price[1]
+      );
+      const date = chartData.payload.prices.map(
+        (price: [number, number]) => price[0]
+      );
+      setChartData(prevData => ({
+        ...prevData,
+        prices: {
+          ...prevData.prices,
+          [index]: prices,
+        },
+        dates: {
+          ...prevData.dates,
+          [index]: date,
+        },
+      }));
+    } catch (err) {
+      console.log(`ERROR FETCHCHARTDATA ON CONVERTER ${err}`);
+    }
+  };
+
+  const handleSelectDay = (day: string) => {
+    setSelectedDays(day);
+  };
+
+  const handleSwap = () => {
+    setIsSwap(!swap);
+    setSelectedCoinsData(prevData => [...prevData.reverse()]);
+  };
+
+  const handleCoinAmount = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setCountAmount(e.target.value);
+  };
+
+  const calculateCoin = (
+    coinOnePrice: number,
+    coinTwoPrice: number
+  ): number => {
+    return (Number(coinAmount) * coinOnePrice) / coinTwoPrice;
+  };
+
+  useEffect(() => {
+    console.log(selectedDays);
+    const fetchCoins = () => {
+      try {
+        const selectedCoinIds = selectedCoins.map(coin => coin.id);
+        const getSelectedCoins = coinsData
+          .filter(c => selectedCoinIds.includes(c.id))
+          .slice(0, 2);
+        setSelectedCoinsData(getSelectedCoins);
+      } catch (err) {
+        console.log(`ERROR IN CONVERT ${err}`);
+      }
+    };
+    fetchCoins();
+    selectedCoins.forEach(({ id }, index) => fetchChartData(id, index));
```

Comment:
Promise.all this thing mate

Reply:
and put this inside the fetchCoins function and use the getSelectedCoins variables for maping